pkg/controller/clusterconfig/clusterconfig_controller.go

```diff
@@ -43,7 +44,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	// Watch for changes to primary resource config.openshift.io/v1/Network
-	err = c.Watch(source.Kind(mgr.GetCache(), &configv1.Network{}), &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{})
+	err = c.Watch(source.Kind[client.Object](mgr.GetCache(), &configv1.Network{}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}))
```

Comment:
For consistency with other files can we use a `crclient` alias for `sigs.k8s.io/controller-runtime/pkg/client`?